### PR TITLE
Add password reset methods to auth-next

### PR DESCRIPTION
**Pull request by avolkovi**

### CODE REVIEW

File: packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseError } from '@firebase/util';
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as sinonChai from 'sinon-chai';
+// import { SinonStub, stub, restore } from 'sinon';
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts:22]: get rid of this?

File: packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+import { mockEndpoint } from '../../../test/api/helper';
+import { mockAuth } from '../../../test/mock_auth';
+import * as mockFetch from '../../../test/mock_fetch';
+import { Endpoint } from '../../api';
+import { ServerError } from '../../api/errors';
+import { Operation } from '../../model/action_code_info';
+// import * as location from '../util/location';
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts:29]: and this?

File: packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+          message: ServerError.INVALID_OOB_CODE
+        }
+      },
+      400
+    );
+    await expect(
+      confirmPasswordReset(mockAuth, oobCode, newPassword)
+    ).to.be.rejectedWith(
+      FirebaseError,
+      'Firebase: The action code is invalid. This can happen if the code is malformed]: expired]: or has already been used. (auth/invalid-action-code).'
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts:165]: what's up with `malformed]: expired ]:` ? are those meant to be `,` ?

File: packages-exp/auth-exp/src/core/strategies/email_and_password.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    operation: response.requestType
+  };
+}
+
+export async function verifyPasswordResetCode(
+  auth: Auth,
+  code: string
+): Promise<string> {
+  const { data } = await checkActionCode(auth, code);
+  return data.email!;
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/src/core/strategies/email_and_password.ts:83]: Throw a comment on here explaining why it's safe to bang-assert this (or consider changing the interface to make it a non-optional field)

File: packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts (hunk 3 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts:165]: good catch! Apparently we had a bunch of typos in the errors map, probably due to bad regexp on my part

File: packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts (hunk 1 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts]: done

File: packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts (hunk 2 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/core/strategies/email_and_password.test.ts]: done

File: packages-exp/auth-exp/src/core/strategies/email_and_password.ts (hunk 4 above)
Comment (avolkovi) [reply packages-exp/auth-exp/src/core/strategies/email_and_password.ts:84]: done

**Pull request merged**